### DISCUSSION TITLE
Filter modal creator dropdown

### DISCUSSION THREAD

**Discussion by Sachin-chaurasiya** [reactplay/react-play#213] [Ideas]
Today we have this simple creator list dropdown which only shows the creator name and does not provide proper context for the creator's Github profile. 

<img width="489" alt="image" src="https://user-images.githubusercontent.com/59080942/169648270-571bf1f7-17f9-42ed-a611-7707564376c3.png">

So I am looking for design suggestions for this, we can take some inspiration from how Github shows the dropdown with profile image, name and username.

<img width="310" alt="image" src="https://user-images.githubusercontent.com/59080942/169648482-e7f426bc-dea5-4ccc-92f9-f49d8efa39b3.png">

Please provide your feedback and suggestions.

Thank You.

**Comment by atapas** [comment]
The design looks great to me. Thanks @Sachin-chaurasiya 

How are you planning to handle the API calls? There may be so many creators, how many calls will be made to get the details?

**Reply by Sachin-chaurasiya** [reply]
@atapas, When the creator submits any play we ask for Github username right? so based on that username we can create a user profile picture URL something like this `https://github.com/{username}.png` 

For example, if I want your Github profile picture, I can quickly get that like this: https://github.com/atapas.png.



Please let me know your thought on this.

**Comment by Sachin-chaurasiya** [comment]
@atapas , Now we are storing plays on the store, so do we have creators' details available as well?

**Reply by Sachin-chaurasiya** [reply]
I can see we have the `user` attribute in play details, and it has a structure like this.

```
{
    "id": "0680f581-6584-4bc4-bbe9-aa7c97567e72",
    "displayName": "Tapas Adhikary",
    "avatarUrl": "https://avatars.githubusercontent.com/u/3633137?v=4"
}
```

**Reply by Sachin-chaurasiya** [reply]
@atapas, Do we have an endpoint available to get the creators list?
cc : @koustov

**Reply by atapas** [reply]
That's right.. the entire details is available.. In fact, the current filter UI puls it using the query. You check the filter code path, it uses the query already, you may have to adjust it to pull the avatarURl if it is not pulling already.

**Reply by Sachin-chaurasiya** [reply]
Thanks, @atapas, I will have a look.
